sys-utils/irqtop.c

```diff
@@ -346,6 +350,10 @@ static void parse_args(	struct irqtop_ctl *ctl,
 		case 'S':
 			ctl->softirq = 1;
 			break;
+		case 't':
+			if (parse_size(optarg, &ctl->threshold, NULL) < 0)
+				errx(EXIT_FAILURE, _("error: --threshold"));
+			break;
```

Comment:
We have prepared function for this pattern:
```C
ctl->threshold = strtosize_or_err(optarg, _("failed to parse threshold"));
```

Reply:
done :)